addons/web/static/src/core/attachment_viewer/attachment_viewer.js

```diff
@@ -8,7 +8,7 @@ import { Component, useExternalListener, useRef, useState } from "@odoo/owl";
  * @extends {Component<Props, Env>}
```

Comment:
_This comment is meant to target the line above but it's not in diff._

It would be nice to take the opportunity to properly document the expected shape of "attachments" to be displayed by the viewer.

It was originally meant to display records from `Attachment` model, but it is more and more given arbitrary objects.

This also goes together with the suggestion from @alexkuhn to rename to "file" viewer or a more generic name.

Reply:
Do you mean to list all the getters and constants from the `Attachment` model? That'd be _a lot_ though.
Like so:
```js
{
    originThread: import("@mail/core/thread_model").Thread,
    isDeletable: boolean,
    displayName: string,
    isText: boolean,
    isPdf: boolean,
    isImage: boolean,
    isUrl: boolean,
    isUrlYoutube: boolean,
    isVideo: boolean,
    isViewable: boolean,
    imageUrl: string,
    defaultSource: string,
    downloadUrl: string,
    _store: import("@mail/core/store_service").Store,
    accessToken: string,
    checksum: string,
    extension: string,
    filename: string,
    id: string,
    mimetype: string,
    name: string,
    originThreadLocalId: string,
    type: string,
    tmpUrl: string,
    url: string,
    uploading: boolean,
    message: import('@mail/core/message_model').Message,
}
```
An alternative would be to only list the ones that show up in the template:
```js
{
     isImage: boolean,
     isPdf: boolean,
     isText: boolean,
     isVideo: boolean,
     displayName: string,
     downloadUrl: string,
     imageUrl: string,
     defaultSource: string,
     isUrlYoutube: boolean,
     mimetype: string,
 }
 ```

Reply:
But then I can't use props validation right? Because for instance `_store` is for sure passed as a prop...

Reply:
Would be nice, as an OWL improvement, that props validation on shape is partial.
@ged-odoo what do you think?